test/test_dot_device.cc

```diff
@@ -0,0 +1,201 @@
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+
+// -----------------------------------------------------------------------------
+template< typename Tx, typename Ty >
+void test_dot_device_work( Params& params, bool run )
+{
+    using namespace testsweeper;
+    using namespace blas;
+    typedef scalar_type<Tx, Ty> scalar_t;
+    typedef real_type<scalar_t> real_t;
+    typedef long long lld;
+
+    // get & mark input values
+    char mode       = params.pointer_mode();
+    int64_t n       = params.dim.n();
+    int64_t incx    = params.incx();
+    int64_t incy    = params.incy();
+    int64_t device  = params.device();
+    int64_t verbose = params.verbose();
+
+    scalar_t  result_host;
+    scalar_t* result = &result_host;
+    scalar_t  result_cblas;
+
+    // mark non-standard output values
+    params.gflops();
+    params.gbytes();
+    params.ref_time();
+    params.ref_gflops();
+    params.ref_gbytes();
+
+    // adjust header to msec
+    params.time.name( "BLAS++\ntime (ms)" );
+    params.ref_time.name( "Ref.\ntime (ms)" );
+
+    if (! run)
+        return;
+
+    if (blas::get_device_count() == 0) {
+        params.msg() = "skipping: no GPU devices or no GPU support";
+        return;
+    }
+
+    // setup
+    size_t size_x = (n - 1) * std::abs(incx) + 1;
+    size_t size_y = (n - 1) * std::abs(incy) + 1;
+    Tx* x    = new Tx[ size_x ];
+    Tx* xref = new Tx[ size_x ];
+    Ty* y    = new Ty[ size_y ];
+    Ty* yref = new Ty[ size_y ];
+
+    // device specifics
+    blas::Queue queue(device, 0);
+    Tx* dx;
+    Ty* dy;
+
+    dx = blas::device_malloc<Tx>(size_x, queue);
+    dy = blas::device_malloc<Ty>(size_y, queue);
+    if (mode == 'd') {
+        result = blas::device_malloc<scalar_t>(1, queue);
+        #if defined( BLAS_HAVE_CUBLAS )
+        cublasSetPointerMode(queue.handle(), CUBLAS_POINTER_MODE_DEVICE);
+        #elif defined( BLAS_HAVE_ROCBLAS )
+        rocblas_set_pointer_mode( queue.handle(), rocblas_pointer_mode_device );
+        #endif
+    }
+
+    // test error exits
+    assert_throw( blas::dot( -1, x, incx, y, incy, result, queue ), blas::Error );
+    assert_throw( blas::dot(  n, x,    0, y, incy, result, queue ), blas::Error );
+    assert_throw( blas::dot(  n, x, incx, y,    0, result, queue ), blas::Error );
+
+    int64_t idist = 1;
+    int iseed[4] = { 0, 0, 0, 1 };
+    lapack_larnv( idist, iseed, size_x, x );
+    lapack_larnv( idist, iseed, size_y, y );
+    cblas_copy( n, x, incx, xref, incx );
+    cblas_copy( n, y, incy, yref, incy );
+
+    blas::device_setvector(n, x, std::abs(incx), dx, std::abs(incx), queue);
+    blas::device_setvector(n, y, std::abs(incy), dy, std::abs(incy), queue);
+    queue.sync();
+
+    if (verbose >= 1) {
+        printf( "\n"
+                "n=%5lld, incx=%5lld, sizex=%10lld, incy=%5lld, sizey=%10lld\n",
+                (lld) n, (lld) incx, (lld) size_x, (lld) incy, (lld) size_y );
```

Comment:
Compare your files with what's in `test_dot.cc`. A number of conventions have changed — one of the reasons to always start from the latest master.
`(lld) n` changes to `llong( n )`.